Parser.py

```diff
@@ -41,35 +54,114 @@ def goto(self, S, x):
                 rightSideOfDot = rightSideOfDot.strip()
```

Comment:
maybe you could create a separate method for finding the left and right side of the dot since you use it multiple times